File: Anul-2/Semestrul I/MAP/Seminar/Seminar-10/Seminar-10/container/AbstractContainer.cs
namespace Seminar_10;

public abstract class AbstractContainer:Container
{    protected Task[] tasks; 
     protected int top;

     public AbstractContainer(int capacity)
     {
         tasks = new Task[capacity];
         top = -1;
     }

     public abstract Task Remove();

    public void Add(Task task)
    {
        if (top < tasks.Length-1)
        {
            top++;
            tasks[top] = task;
        }
        else throw new InvalidOperationException("Container is full");
    }

    public int Size()
    {
        return top + 1;
    }

    public bool IsEmpty()
    {
        return top == -1;

    }

    public Task GetTopTask()
    {
        return tasks[top];
    }

    public int Top
    {
        get => top;
    }
}

File: Anul-2/Semestrul I/MAP/Laborator/LucruInClasa/lab2_schelet (1)/sem1_2/model/AbstractContainer.java
package sem1_2.model;

import java.util.ArrayList;
import java.util.List;

public abstract class AbstractContainer implements Container {
    protected ArrayList<Task> list;
    AbstractContainer(){
        this.list=new ArrayList<>();
    }
    @Override
    public int size(){
        return list.size();
    }
    @Override
    public boolean isEmpty(){
        if(list.isEmpty()){
            return true;
        }
        else
            return false;

    };
    @Override
    public void add(Task task) {
        list.add(task);
    }
}

